sklearn/neural_network/tests/test_mlp.py

```diff
@@ -752,7 +752,7 @@ def test_warm_start_full_iteration(MLPEstimator):
     clf.fit(X, y)
     assert max_iter == clf.n_iter_
     clf.fit(X, y)
-    assert 2 * max_iter == clf.n_iter_
+    assert max_iter == clf.n_iter_
```

Comment:
@glemaitre Given that we test this behavior, I think this backward incompatible change is too big to be in a bug fix release (1.2.2).

The safe choice is to go through a deprecation cycle, but we can also consider this a bug fix for the 1.3 release.

Reply:
I think it's a bug fix because it didn't follow the expectation of `warm_start`, probably because our expectation were not clear enough. So I think that we don't need a deprecation cycle. But I'm fine with targeting 1.3 instead of 1.2.2.

Reply:
I am fine as well for delaying to 1.3 but I am surely convinced that this is a bug :)